### DISCUSSION TITLE
Removal of tracedSvg and TRACED_SVG

### DISCUSSION THREAD

**Discussion by Khaledgarbaya** [gatsbyjs/gatsby#37104] [Umbrella Discussions]
Due to licensing requirements, we are removing the [Potrace library](https://potrace.sourceforge.net/) usage from Gatsby 5, specifically starting with **v5.2**.

Potrace is used in Gatsby's image transformations, specifically for creating an SVG placeholder. For example, in `gatsby-plugin-image` you were able to use it with `tracedSvg` and `TRACED_SVG` respectively.

These are the affected libraries/plugins:

- `gatsby-plugin-sharp@5.2.0`
- `gatsby-transformer-sharp@5.2.0`
- `gatsby-plugin-utils@4.2.0`
- `gatsby-remark-images@7.2.0`

There will be **no API-breaking changes**, your builds will continue to work. However, whenever you use the `tracedSvg` feature Gatsby automatically falls back to the default placeholder style now. In most instances, this is `DOMINANT_COLOR`. You'll see a detailed warning message in your CLI about this, too. 

The Potrace library has a GPL license which would require Gatsby to be also licensed as GPL. We don't have any plans to change our current MIT license to GPL therefore we are removing Potrace.

You can read more about MIT vs GPL license [at choosealicense](https://choosealicense.com/)

**Comment by avodynamics** [comment]
Are there any plans to try and get this functionality back through a different method/library?
This feels like a big loss to me, but I haven't seen much discussion on it.

**Reply by DSchau** [reply]
Not at this time. We will investigate further in 2023 to see what's possible and we appreciate your feedback!

**Reply by shubhamdhingra007** [reply]
Hi @DSchau! Using DOMINANT_COLOR as a replacement is fine, but we didn't expect upgrading to Gatsby@5 will cause visual behaviour updates(we have image gallery pages). I mean at least a way to achieve that could have been provided and would have been appreciated.
 
If Gatsby wasn't providing this feature, that would have made sense. But for us to upgrade Gatsby version for optimizations and benefits and then change the internal independent implementations to maintain the same behaviour is costly.

Now my way of thinking is we could have opted for any other 3rd party library for this rather than using the fancy features of Gatsby. Even if they update or have breaking changes in future versions, we will need to care only about that. But in the case of Gatsby upgrade, we need to ensure all the new prerequisites like react@18(which itself is a breaking change) node@18, migrate all stuff compatible to Gatsy@5(including this issue), test them; which makes it cumbersome, and extreme time consuming.
